library/src/main/java/com/datatorrent/lib/window/WindowOption.java

```diff
@@ -0,0 +1,143 @@
+package com.datatorrent.lib.window;
+
+import java.util.LinkedList;
+import java.util.List;
+
+/**
+ *
+ */
+public abstract class WindowOption
+{
+
+  private TriggerOption[] triggerOptions;
+
+  private AccumulationMode accumulationMode = AccumulationMode.ACCUMULATE;
+
+  private List<Object[]> lateness = new LinkedList<>();
```

Comment:
Does it make sense for this to be a set, rather than a list? The same window may be added multiple times but we should only calcualte it once.